features/step_definitions/table_steps.rb

```diff
@@ -21,7 +21,7 @@
 end
 
 # TODO: can we use "has_table?" instead of this complex step?
-Then "(I )(they )should see (the )following table(:)" do |expected|
+Then /^(?:I )?(?:they )?should see (?:the )?following table( with exact columns)?:?$/ do |exact_columns, expected|
```

Comment:
The step was change to avoid touching all the existing scenarios.
I couldn't find a way to add a parameter type within an optional block... In fact, I tried, but cucumber said it was not possible.

Reply:
Not possible, I've wanted to do the same many times.
However, this is a bit redundant as there is already a way to do this:
```cucumber
Then the table has a column "Plan"
But the table does not have a column "Plan"
```